src/pure.js

```diff
@@ -207,6 +207,14 @@ function render(
     wrapper,
   } = {},
 ) {
+  if (legacyRoot && typeof ReactDOM.render !== 'function') {
+    const error = new Error(
+      '`legacyRoot: true` is not supported in this version of React. Please use React 18 instead.',
```

Comment:
Perhaps this message could be more encouraging update, rather than encouraging tech debt?

Reply:
Good point. Do you have a suggestion for a better warning?

Reply:
My thoughts:
* `legacyRoot` is only relevant for earlier versions of React
* If you're testing with React 19, there's a high chance you are also USING React 19, then it'd be best to just remove it
* If you're testing with React 19 but somehow you are NOT using React 19 yourself (?!), you may need help with migration.

Given that, my take:

> Warning: legacyRoot is no longer supported in React 19. If your app runs React 19 or later, you should remove this flag. If your app runs React 18 or earlier, and you need help with upgrading your app, visit https://reactjs.org/link/switch-to-createroot.

Reply:
Makes sense: https://github.com/testing-library/react-testing-library/pull/1301. The link is not that applicable since you wouldn't need to change any callsites of `render` from RTL.